### PR TITLE
PERF-5094 Update toolchain version for libmongocrypt Range V2 support

### PR DESCRIPTION
**Pull request by marksg07**
**Jira Ticket:** PERF-5094

**Whats Changed:**  
Updates the dependency for libmongocrypt to a recent version for Range V2 support. This is necessary for running the performance tests described in the linked ticket. When the next version of libmongocrypt releases, we plan to do another update to the toolchain version so that we don't have a git-hash version in the toolchain; see https://jira.mongodb.org/browse/PERF-5064

**Related PRs:**   
PR for toolchain change: https://github.com/10gen/vcpkg/pull/41

### CODE REVIEW

**Comment by lgfang** [comment]
I believe the relevant ticket is PERF-5064, not PERF-5094. Have I overlooked anything?

**Review by lgfang — APPROVED** [review]
LGTM although I'd prefer a waterfall build ID (will be available after the corresponding vcpkg change is merged in).

In case this change affects performance test results, please also get an approval from the product performance team.

**Review by jimoleary — APPROVED** [review]
LGTM but do we know if this will cause any change points?

I was about to ask about the **patch_**  but you covered with:

```
When the next version of libmongocrypt releases, we plan to do another update to the toolchain version so that we don't have a git-hash version in the toolchain; see https://jira.mongodb.org/browse/PERF-5064
```

**Comment by marksg07** [comment]
@jimoleary I ran a sys-perf test with this patch: https://spruce.mongodb.com/version/66159eb2ce99a60007ab8ce4/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC I'm not super familiar with the perf tooling, so I don't know if there are significant change points, but we don't expect there to be. 

@lgfang Is there an @ I can add as a reviewer for the product performance team?

**Comment by marksg07** [comment]
https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=04967c72-242e-4933-a603-15baeb0562e4 Perf analysis

**Comment by thessem** [comment]
@marksg07 The @ is @mongodb/performance in GitHub and `@product-perf` in slack (for `ask-devprod-performance`), however this the team that @jimoleary is on so you're already speaking to them.

**Comment by jimoleary** [comment]
@marksg07 It looks like you only ran Shard Lite FLE ARM AWS 2023-11 tasks ... you might want to run a broader set of variants to have a better idea.

That being said, the possible regressed task metrics are all only on setup (shardCollection , createIndex or insert) phases.

**Comment by marksg07** [comment]
@jimoleary I ran a bunch more tests, do they look good to you? It seems like a lot of the regressions are in setup tasks. Thanks! https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=13acd534-7db7-4082-be72-c2e2b2554375

**Review by jimoleary — APPROVED** [review]
LGTM but it looks like this change may cause some change points.

**Pull request merged**
